clients/java/src/main/java/io/zeebe/client/impl/ZeebeClientCloudBuilderImpl.java

```diff
@@ -0,0 +1,219 @@
+      withClusterId(properties.getProperty(ClientProperties.CLOUD_CLUSTER_ID));
+    }
+    if (properties.containsKey(ClientProperties.CLOUD_CLIENT_ID)) {
+      withClientId(properties.getProperty(ClientProperties.CLOUD_CLIENT_ID));
+    }
+    if (properties.containsKey(ClientProperties.CLOUD_CLIENT_SECRET)) {
+      withClientSecret(properties.getProperty(ClientProperties.CLOUD_CLIENT_SECRET));
+    }
+    return innerBuilder.withProperties(properties);
+  }
+
+  @Override
+  public ZeebeClientBuilder defaultJobWorkerMaxJobsActive(final int maxJobsActive) {
+    return innerBuilder.defaultJobWorkerMaxJobsActive(maxJobsActive);
+  }
+
+  @Override
+  public ZeebeClientBuilder numJobWorkerExecutionThreads(final int numThreads) {
+    return innerBuilder.numJobWorkerExecutionThreads(numThreads);
+  }
+
+  @Override
+  public ZeebeClientBuilder defaultJobWorkerName(final String workerName) {
+    return innerBuilder.defaultJobWorkerName(workerName);
+  }
+
+  @Override
+  public ZeebeClientBuilder defaultJobTimeout(final Duration timeout) {
+    return innerBuilder.defaultJobTimeout(timeout);
+  }
+
+  @Override
+  public ZeebeClientBuilder defaultJobPollInterval(final Duration pollInterval) {
+    return innerBuilder.defaultJobPollInterval(pollInterval);
+  }
+
+  @Override
+  public ZeebeClientBuilder defaultMessageTimeToLive(final Duration timeToLive) {
+    return innerBuilder.defaultMessageTimeToLive(timeToLive);
+  }
+
+  @Override
+  public ZeebeClientBuilder defaultRequestTimeout(final Duration requestTimeout) {
+    return innerBuilder.defaultRequestTimeout(requestTimeout);
+  }
+
+  @Override
+  public ZeebeClientBuilder caCertificatePath(final String certificatePath) {
+    return innerBuilder.caCertificatePath(certificatePath);
+  }
+
+  @Override
+  public ZeebeClientBuilder keepAlive(final Duration keepAlive) {
+    return innerBuilder.keepAlive(keepAlive);
+  }
+
+  @Override
+  public ZeebeClientBuilder withInterceptors(final ClientInterceptor... interceptor) {
+    return innerBuilder.withInterceptors(interceptor);
+  }
+
+  @Override
+  public ZeebeClientBuilder withJsonMapper(final JsonMapper jsonMapper) {
+    return innerBuilder.withJsonMapper(jsonMapper);
+  }
+
+  @Override
+  public ZeebeClient build() {
+    innerBuilder.gatewayAddress(determineGatewayAddress());
+    innerBuilder.credentialsProvider(determineCredentialsProvider());
+    return innerBuilder.build();
+  }
+
+  private String determineGatewayAddress() {
+    if (isNeedToSetCloudGatewayAddress()) {
+      ensureNotNull("cluster id", clusterId);
+      return String.format("%s.%s:443", clusterId, BASE_ADDRESS);
+    } else {
+      Loggers.LOGGER.debug(
+          "Expected that the gateway address is not set, "
+              + "but got this gateway address: {} set in the client cloud builder; "
+              + "should be not set if you are trying to connect to the Camunda Cloud.",
+          innerBuilder.getGatewayAddress());
```

Comment:
We think it's fine to overwrite the default. So let's use something similar here:
`Expected to use 'cluster id' to set gateway address in the client cloud builder, but overwriting with explicitly defined gateway address`